noxfile.py

```diff
@@ -36,6 +36,38 @@ def set_environment_variables(env_dict, session):
         session.env[key] = value
 
 
+def install_PyBaMM(session, extras):
```

Comment:
This looks redundant at the first look (and is maybe overcomplicating things). We can simply add an `and` condition in `if-else` blocks to check for Python 3.12

Reply:
The issue with not having this function is that there will be too many `if-else` blocks to handle. They would work for tests on PRs, but we are also using `nox` for local development besides CI

Reply:
If the issue is just for JAX and ODEs, can we just make them a separate install? For instance, pull them out of dev dependencies and just make an additional extras group for modules with potential install difficulties?

Reply:
`jax` and `odes` are separate installs at this moment with their own groups; it wouldn't make sense to add another extra set with both of those and then not document that for users. I don't think there is currently a provision for a "semi-private" set of extras of the type `[_odes]`. The problem here is that we can't tailor the noxfile just for our CI needs because it is also being used locally by developers: a user on, say, macOS should be able to install `jax` through any `nox` session if they want instead of restricting that functionality to a particular platform

I can make the change and add if-else blocks for every session but it won't conform to DRY practices